File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
import java.math.BigInteger;
class Solution {
    public static BigInteger fac(int n) {
        BigInteger f = BigInteger.ONE;
        for (int i = 2; i < n + 1; i++){
            f = f.multiply(BigInteger.valueOf(i));
        }
        return f;
    }

    public static BigInteger solution(int balls, int share) {
        if (balls == share){
            return BigInteger.ONE;
        }
        else{
            return fac(balls).divide(fac(balls - share).multiply(fac(share)));
        }
}
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.go
package main

func gcd64(a, b int64) int64 {
    for b != 0 {
        a, b = b, a%b
    }
    return a
}

func solution(balls int, share int) int {
    n, r := balls, share
    if r > n-r { 
        r = n - r
    }

    res := int64(1)
    for i := 1; i <= r; i++ {
        a := int64(n - r + i) 
        b := int64(i)         

      
        g := gcd64(a, b)
        a /= g
        b /= g

     
        g = gcd64(res, b)
        res /= g
        b /= g 

    
        res *= a
        
    }

    return int(res)
}